### ISSUE TITLE
Pyright error with tuple slice

### ISSUE THREAD

**Issue by sa-mendez** [microsoft/pyright#5095]
**Describe the bug**
Getting the following error from pyright, even though the code run fine in the interpreter:
``
 error: Argument of type "tuple[slice, Literal[0]]" cannot be assigned to parameter "__s" of type "slice" in function "__ge
titem__"
    "tuple[slice, Literal[0]]" is incompatible with "slice" (reportGeneralTypeIssues)
``

**To Reproduce**
Run the included code through pyright.  

**Expected behavior**
No error should be shown.
However I am not sure if this is by design or not, but I am asking because it runs perfectly fine in the repl

**Screenshots or Code**
The error occurs when setting foo
```python
from sklearn.datasets import make_circles

X, y = make_circles(n_samples=10, noise=0.03, random_state=42)

foo = X[:, 0]
```

**VS Code extension or command-line**
pyright version 1.1.307
python version 3.10.11
scikit-learn (sklearn) vervsion 1.2.2 

**Additional context**
N/A

**Comment by erictraut** [comment]
You're using the `sklearn` library, which doesn't have any type information. I presume that you don't have any type stubs installed for sklearn. In this case, pyright will do its best to infer types from the code, but type inference can produce incorrect results. In this case, the return type of `make_circles` is a complex tuple type. Once unpacked, the type of `X` evaluates to `Unknown | list[Unknown] | ndarray[Any, dtype[floating[Any]]] | ndarray[Any, dtype[Any]]`. 

If the type of `X` is `list`, then providing a tuple within a slice will result in a runtime exception because `list` doesn't support such slices. That's what the error message is telling you. So pyright is working as designed here.

You have a few options:
1. Set the value of `useLibraryCodeForTypes` to false in your pyright configuration file. This tells it to treat all symbols from untyped libraries (like sklearn) as Unknown. You'll lose all type analysis and language server features (e.g. completion suggestions) for these symbols, but you'll eliminate any false positive errors.
2. Use an `assert` statement to assert the type of `X`. Pyright will then narrow the type from the original union to a narrower type (e.g. `ndarray`).
3. Use a `cast` call to cast the type of `X` to the desired type, effectively overriding the type evaluation logic.
4. Use a `# type: ignore` comment to suppress the type violation error on that line.

**Comment by sa-mendez** [comment]
Ok.  Thank you for the thorough explanation.
